pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"io"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+type PodSandboxID string
+
+// PodSandboxManager provides basic operations to create/delete and examine the
+// PodSandboxes.
+type PodSandboxManager interface {
+	// Create creates a sandbox based on the given config, and returns the
+	//  of the new sandbox.
+	Create(config *PodSandboxConfig) (PodSandboxID, error)
+	// Delete deletes the sandbox by its ID. If there are any running
+	// containers in the sandbox, they will be terminated as a side-effect.
+	Delete(id PodSandboxID) error
+	// List lists existing sandboxes, filtered by the given PodSandboxFilter.
+	List(filter PodSandboxFilter) []PodSandboxListItem
+	// Status gets the status of the sandbox by ID.
+	Status(id PodSandboxID) (PodSandboxStatus, error)
+	// PortForward copies the data from the port in the PodSandBox to the
+	// stream.
+	// TODO: Should PortForward be in a separate interface.
+	PortForward(id PodSandboxID, port uint16, stream io.ReadWriteCloser) error
+}
+
+// PodSandboxConfig holds all the required and optional fields for creating a
+// sandbox.
+type PodSandboxConfig struct {
```

Comment:
Passing the pod would also help in the hyper case of needing mounts of all containers up-front without adding a mounts field that other runtimes may not need (but rather a pod field that other runtimes may not need :)

I don't see why the sandbox manager knowing what containers it's creating a sandbox for is an issue; most could ignore it, some could make some optimizations perhaps (like loading optional modules early as mentioned above, or warming the image cache in some manner).
I understand that there's an abstraction here that we don't want to break, but I don't think this breaks the important part of the abstractions; the separation between the sandbox and containers lifecycles.